src/diagnostics/MOM_debugging.F90

```diff
@@ -837,67 +838,101 @@ end subroutine chksum_vec_A2d
+                              ! call [S H L2 ~> ppt m3 or ppt kg] or [ppt m3] or [ppt kg]
   ! Local variables
   logical, save :: firstCall = .true.
-  real, dimension(HI%isc:HI%iec, HI%jsc:HI%jec) :: tmp_for_sum ! The volume of each column [m3]
-  real :: thisH, delH  ! The total ocean volume and the change from the last call [m3]
-  real :: thisT, delT  ! The current total volume integrated temperature and the change from the last call [degC m3]
-  real :: thisS, delS  ! The current total volume integrated salinity and the change from the last call [ppt m3]
+  real :: tmp_for_sum(HI%isc:HI%iec, HI%jsc:HI%jec) ! The volume of each column [H L2 ~> m3 or kg] or [m3] or [kg]
+  real :: thisH, delH  ! The total ocean volume and the change from the last call [H L2 ~> m3 or kg] or [m3] or [kg]
+  real :: thisT, delT  ! The current total volume integrated temperature and the change from the last
+                       ! call [C H L2 ~> degC m3 or degC kg] or [degC m3] or [degC kg]
+  real :: thisS, delS  ! The current total volume integrated salinity and the change from the last
+                       ! call [S H L2 ~> ppt m3 or ppt kg] or [ppt m3] or [ppt kg]
+  real :: H_unscale    ! A constant that translates thickness units to its MKS units (m or kg m-2) based on
+                       ! whether the model is Boussinesq [m H-1 ~> 1] or not [kg m-2 H-1 ~> 1]
+  real :: HL2_unscale  ! An overall unscaling factor for cell mass or volume [m3 H-1 L-2 ~> 1] or [kg H-1 L-2 ~> 1]
+  real :: T_unscale    ! An overall unscaling factor for cell-integrated temperature [degC m3 C-1 H-1 L-2 ~> 1] or
+                       ! [degC kg C-1 H-1 L-2 ~> 1]
+  real :: S_unscale    ! An overall unscaling factor for cell-integrated salinity [ppt m3 S-1 H-1 L-2 ~> 1] or
+                       ! [ppt kg S-1 H-1 L-2 ~> 1]
   integer :: i, j, k, nz
 
+  H_unscale = 1.0 ; if (present(H_to_mks)) H_unscale = H_to_mks
+  if (present(US)) then
+    HL2_unscale = US%L_to_m**2 * H_unscale
+    T_unscale = US%C_to_degC * HL2_unscale ; S_unscale = US%S_to_ppt * HL2_unscale
+  else
+    HL2_unscale = H_unscale
+    T_unscale = HL2_unscale ; S_unscale = HL2_unscale
+  endif
+
   nz = size(hThick,3)
   tmp_for_sum(:,:) = 0.0
   do k=1,nz ; do j=HI%jsc,HI%jec ; do i=HI%isc,HI%iec
     tmp_for_sum(i,j) = tmp_for_sum(i,j) + hThick(i,j,k) * areaT(i,j)
   enddo ; enddo ; enddo
-  thisH = reproducing_sum(tmp_for_sum)
-  thisT = totalStuff(HI, hThick, areaT, temperature)
-  thisS = totalStuff(HI, hThick, areaT, salinity)
+  thisH = reproducing_sum(tmp_for_sum, unscale=HL2_unscale)
+  thisT = totalStuff(HI, hThick, areaT, temperature, unscale=T_unscale)
+  thisS = totalStuff(HI, hThick, areaT, salinity, unscale=S_unscale)
 
   if (is_root_pe()) then
     if (firstCall) then
       totalH = thisH ; totalT = thisT ; totalS = thisS
-      write(0,*) 'Totals H,T,S:',thisH,thisT,thisS,' ',mesg
+      write(0,*) 'Totals H,T,S:', thisH*HL2_unscale, thisT*T_unscale, thisS*S_unscale, ' ', mesg
```

Comment:
Not necessarily part of this PR, but it's nonstandard to assume that unit 0 is stdout.  Probably simplest to just use `print *`?

Reply:
I have changed this to `write(stdout,*)` where `stdout` comes from the `MOM_io` module, similarly to what is done elsewhere.